ngraph/core/src/op/if.cpp

```diff
@@ -0,0 +1,323 @@
+    : op::util::MultiSubGraphOp(values, 2)
+{
+}
+
+op::v8::If::If(const Output<Node>& execution_condition)
+    : If()
+{
+    set_argument(0, execution_condition);
+}
+
+ngraph::PartialShape resolve_shape(ngraph::PartialShape then_pshape,
+                                   ngraph::PartialShape else_pshape)
+{
+    auto then_rank = then_pshape.rank();
+    auto else_rank = else_pshape.rank();
+    if (then_rank.is_dynamic() || else_rank.is_dynamic() ||
+        then_rank.get_length() != else_rank.get_length())
+    {
+        return ngraph::PartialShape::dynamic(ngraph::Rank::dynamic());
+    }
+    std::vector<Dimension> new_dims;
+    for (auto then_it = then_pshape.cbegin(), else_it = else_pshape.cbegin();
+         then_it != then_pshape.cend();
+         then_it++, else_it++)
+    {
+        if ((*then_it).is_dynamic() || (*else_it).is_dynamic())
+        {
+            new_dims.push_back(Dimension::dynamic());
+        }
+        else if (*then_it == *else_it)
+        {
+            new_dims.push_back(Dimension(*then_it));
+        }
+        else
+        {
+            auto dim_min = std::min((*then_it).get_min_length(), (*else_it).get_min_length());
+            auto dim_max = std::max((*then_it).get_min_length(), (*else_it).get_min_length());
+            new_dims.push_back(Dimension(dim_min, dim_max));
+        }
+    }
+
+    return PartialShape(new_dims);
+}
+
+bool op::v8::If::visit_attributes(AttributeVisitor& visitor)
+{
+    NGRAPH_OP_SCOPE(v8_If_visit_attributes);
+    m_bodies[then_body_index] =
+        std::make_shared<ngraph::Function>(OutputVector{}, ParameterVector{}, "then_branch");
+    m_bodies[else_body_index] =
+        std::make_shared<ngraph::Function>(OutputVector{}, ParameterVector{}, "else_branch");
+    visitor.on_attribute("then_body", m_bodies[then_body_index]);
+    visitor.on_attribute("else_body", m_bodies[else_body_index]);
+    visitor.on_attribute("then_inputs", m_input_descriptions[then_body_index]);
+    visitor.on_attribute("else_inputs", m_input_descriptions[else_body_index]);
+    visitor.on_attribute("then_outputs", m_output_descriptions[then_body_index]);
+    visitor.on_attribute("else_outputs", m_output_descriptions[else_body_index]);
+    return true;
+}
+
+void op::v8::If::validate_and_infer_type_body(
+    std::shared_ptr<Function> body,
+    ngraph::op::util::MultiSubgraphInputDescriptionVector& input_descriptors)
+{
+    auto layer_id_map = std::map<size_t, std::shared_ptr<op::Parameter>>();
+    for (auto param : body->get_parameters())
+    {
+        layer_id_map.insert({param->get_instance_id(), param});
+    }
+    for (const auto& input_description : input_descriptors)
+    {
+        auto index = input_description->m_input_index;
+
+        auto body_parameter = body->get_parameters().at(input_description->m_body_parameter_index);
+        auto input_partial_shape = inputs().at(index).get_source_output().get_partial_shape();
+        if (input_partial_shape.is_static())
```

Comment:
`body_parameter->set_partial_shape(input_partial_shape);` instead of this if-else